From cbba8cb3b86ea3d7e976ec456589d8f4fc081843 Mon Sep 17 00:00:00 2001
From: "Langer, Thomas" <thomas.langer@intel.com>
Date: Fri, 24 Jul 2020 18:58:13 +0200
Subject: [PATCH] PONRTSYS-7812: Fix BBF 247 test cases XGSPON

Revert "PONRTSYS-7569: ETH Driver - update PCS settings when updating PHY speed"
This reverts commit dc0d18f812b4cdc490e6c79409ff2473b3caa201.

The change is breaking the link between MAC and PHY if we only try to configure
"Advertised link modes", as it will configure the speed on xpcs with the
current speed "status", maybe not actively set by the caller (just left
as the "get" call was reporting it).
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index 113183414128..f6d618004e0e 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -89,9 +89,6 @@ static int phy_netdevice_event(struct notifier_block *nb, unsigned long action,
 static int xrx500_phy_connect(struct net_device *dev, struct xrx500_port *port);
 static int ltq_gsw_pmac_init(void);
 
-static int serdes_ethtool_set_link_ksettings(struct net_device *dev,
-				      const struct ethtool_link_ksettings *cmd);
-
 /**
  *  Datapath directpath functions
  */
@@ -237,8 +234,6 @@ static int get_link_ksettings(struct net_device *dev,
 static int set_link_ksettings(struct net_device *dev,
 			      const struct ethtool_link_ksettings *cmd)
 {
-	int ret;
-
 	/* Linux PHY framework does not yet support 2500baseT,
 	 * so we need to configure manually here.
 	 */
@@ -250,10 +245,6 @@ static int set_link_ksettings(struct net_device *dev,
 		__clear_bit(ETHTOOL_LINK_MODE_2500baseT_Full_BIT,
 			    dev->phydev->extended_advertising);
 
-	ret = serdes_ethtool_set_link_ksettings(dev, cmd);
-	if (ret)
-		pr_debug("debug: serdes_ethtool_set_link_ksettings return %d\n", ret);
-
 	return phy_ethtool_set_link_ksettings(dev, cmd);
 }
 
